### DISCUSSION TITLE
Add a flag to AllowNegativeRates only during the day

### DISCUSSION THREAD

**Discussion by zigomatichub** [jomjol/AI-on-the-edge-device#1707] [General]
### The Feature

In the power metering measurement usecase, I would like to limit the negative rates when it's needed.
For instance between sunrise and sunset ( + add an offset of xx min ) where solar panels injection is possible.

Then I will not get any negative rate and limit the possibility of errors. 

I see an code example that may help:
https://github.com/Aircoookie/WLED/blob/main/wled00/ntp.cpp

**Comment by caco3** [comment]
I think this better should be solved in a homeautomation system!

Eg. Homeassistant has sensors which allows to do actions based on the current sunrise/fall

**Comment by zigomatichub** [comment]
Hello @caco3 
 I was thinking some extra code can eliminate from the source but indeed from HA it's possible. 
 I guess one more sensor is needed to store the value and check before if negative value, then discard that update received from AI jomjol. 
 I don't know the effort to implement that. I also see some issues asking about NTP format where in the code example, it may give a good alternative to may go to another implementation than posix format. (I'm fine with the actual implementation , this is already awesome)
